src/drivers/intel/cavs/idc.c

```diff
@@ -177,6 +179,37 @@ static int idc_wait_in_blocking_mode(uint32_t target_core)
 	return 0;
 }
 
+/**
+ * \brief Waits until slave core powers up.
+ * \param[in] core Id of the core powering up.
+ * \return Error code.
+ */
+static int idc_wait_for_power_up(uint32_t core)
+{
+	uint32_t offset = PLATFORM_TRACE_SLAVE_CORE(core);
+	struct timer *timer = timer_get();
+	uint64_t deadline;
+
+	deadline = platform_timer_get(timer) +
+		clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1) *
+		IDC_TIMEOUT / 1000;
+
+	while (mailbox_sw_reg_read(offset) != TRACE_BOOT_PLATFORM) {
```

Comment:
this function seems to be identical to idc_wait_in_blocking_mode() with only a different test condition. Can we merge them into something like `idc_wait_core_status(core, bool (*test_fn)(core))`?

Reply:
Done.